core/src/main/scala/kafka/coordinator/transaction/TransactionStateManager.scala

```diff
@@ -113,6 +118,16 @@ class TransactionStateManager(brokerId: Int,
   def getTransactionState(transactionalId: String): Option[CoordinatorEpochAndTxnMetadata] = {
     val partitionId = partitionFor(transactionalId)
 
+    // we only need to check leaving partition set but not loading partition set since there are three possible cases:
+    //    1) it is not in the loading partitions set, hence safe to return NONE
+    //    2) it is in the loading partitions with a smaller epoch, hence safe to return NONE
+    //    3) it is in the loading partition with a larger epoch, return NONE is also fine as it
+    //       indicates the metadata is not exist at least for now.
+    //
+    //    4) it is NOT possible to be in the loading partition with the same epoch
+    if (leavingPartitions.exists(_.txnPartitionId == partitionId))
+      return None
```

Comment:
Hmm, not sure if it's always ok to return None in this case since the caller in TransactionCoordinator.handleInitProducerId() will treat None as no entry in transactionMetadataCache and will try to create a new entry.